### PR TITLE
clang-tidy: C array to std::array conversions

### PR DESCRIPTION
**Pull request by neheb**
Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

**Comment by neheb** [comment]
Funny. This exposed bugs.

File: src/exif.cpp [hunk 1]
```diff
@@ -545,12 +542,13 @@ WriteMethod ExifParser::encode(Blob& blob, const byte* pData, size_t size, ByteO
...
-                                       subImage6Id, subImage7Id, subImage8Id, subImage9Id, subThumb1Id,
-                                       panaRawId,   ifd2Id,      ifd3Id};
+  static constexpr auto filteredIfds = std::array{
+      subImage1Id, subImage2Id, subImage3Id, subImage4Id, subImage5Id, subImage6Id, subImage7Id,
+      subImage8Id, subImage9Id, subThumb1Id, panaRawId,   ifd2Id,      ifd3Id,
+  };
   for (auto&& filteredIfd : filteredIfds) {
 #ifdef EXIV2_DEBUG_MESSAGES
-    std::cerr << "Warning: Exif IFD " << filteredIfds << " not encoded\n";
+    std::cerr << "Warning: Exif IFD " << filteredIfd << " not encoded\n";
```
Comment (piponazo) [inline src/exif.cpp:551]: LoL, I wonder what was happening here before 🙉

**Review by piponazo — APPROVED** [review]
LGTM 👍

File: src/exif.cpp (hunk 1 above)
Comment (neheb) [reply src/exif.cpp:551]: Probably same thing since there's only one member.

**Pull request merged**